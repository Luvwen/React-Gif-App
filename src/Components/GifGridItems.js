import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItems = ({ title, url }) => {

  return (
    <div className={'card animate__animated animate__fadeIn'}>
      <img src={url} alt={title}></img>
      <p>{title}</p>
    </div>
  );
};

GifGridItems.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
