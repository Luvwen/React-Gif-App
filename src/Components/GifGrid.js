import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItems } from './GifGridItems';

export const GifGrid = ({ category }) => {
    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3 className="category animate__animated animate__fadeIn">{category}</h3>
            <div className={'card-grid'}>
                {loading && <p className="animate__animated animate__flash">Loading</p>}
                {images.map(img => (
                    <GifGridItems key={img.id} {...img} />
                ))}
            </div>
        </>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
};
