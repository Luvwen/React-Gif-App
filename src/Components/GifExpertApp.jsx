import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

export const GifExpertApp = ({ defaultCategories=[] }) => {
  const [categories, setCategories] = useState(
    defaultCategories
  );

  return (
    <div className="container-card">
      <h2>Gif Expert App</h2>
      <AddCategory setCategories={setCategories} />
      <ol>
        {categories.map((category) => <GifGrid 
        key={ category }
        category={ category }/>)}
      </ol>
    </div>
  );
};
