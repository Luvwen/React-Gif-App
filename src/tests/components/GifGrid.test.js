import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { GifGrid } from '../../Components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Prueba en GifGrid', () => {
    const category = 'Hola mundo';

    test('debe hacer match el snapshot', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan las imÃ¡genes useFetchGif', () => {
        const gifs = [
            {
                id: 'ABC',
                url: 'https://localhost/galery/sasuke.jpg',
                title: 'Sasuke',
            },
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItems').length).toBe(gifs.length);
    });
});
