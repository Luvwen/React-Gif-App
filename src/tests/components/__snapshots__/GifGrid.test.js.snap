// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en GifGrid debe de mostrar items cuando se cargan las im√°genes useFetchGif 1`] = `
<Fragment>
  <h3
    className="category animate__animated animate__fadeIn"
  >
    Hola mundo
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItems
      id="ABC"
      key="ABC"
      title="Sasuke"
      url="https://localhost/galery/sasuke.jpg"
    />
  </div>
</Fragment>
`;

exports[`Prueba en GifGrid debe hacer match el snapshot 1`] = `
<Fragment>
  <h3
    className="category animate__animated animate__fadeIn"
  >
    Hola mundo
  </h3>
  <div
    className="card-grid"
  >
    <p
      className="animate__animated animate__flash"
    >
      Loading
    </p>
  </div>
</Fragment>
`;
